src/core.rs

```diff
@@ -107,7 +108,19 @@ impl Core {
         info!("Updating current view");
 
         if let Some(view) = self.views.get_mut(&self.current_view) {
-            view.update(update)
+            view.update_lines(update)
+        } else {
+            error!("View {} not found", &self.current_view);
+            bail!(ErrorKind::UpdateError);
+        }
+    }
+
+    pub fn scroll_to(&mut self, cursor: &Cursor) -> Result<()> {
+        info!("Updating cursor position");
+
+        if let Some(view) = self.views.get_mut(&self.current_view) {
```

Comment:
actually this does not compile, because `get_view_mut` borrow `self` mutably:
```rust
        // borrows &mut self
        if let Some(view) = self.get_view_mut() {
            view.update_cursor(cursor);
            Ok(())
        } else {
           // try to borrow &self but it's already borrowed mutable
            error!("View {} not found", &self.current_view);
            bail!(ErrorKind::UpdateError);
        }
```
so I'm going to keep that. I think it's fine within the Impl itself.